dwdatareader/__init__.py

```diff
@@ -1,171 +1,347 @@
-"""Python module to interact with Dewesoft DWDataReaderLib.dll
+"""Python module that wraps Dewesoft DWDataReaderLib.dll for interactive use with Pyton
 
-@author: shelmore and costerwisch
+@author: shelmore, costerwisch and Bojan023
 
 Example usage:
 import dwdatareader as dw
 with dw.open('myfile.d7d') as f:
     print(f.info)
     ch1 = f['chname1'].series()
-    ch1.plot()
     for ch in f.values():
         print(ch.name, ch.series().mean())
 """
-__all__ = ['DWError', 'DWFile', 'getVersion']
-__version__ = '0.16.0'
 
-DLL = None # module variable accessible to other classes
-encoding = 'ISO-8859-1'  # default encoding
-
-import os
 import ctypes
-
-try:
-    from collections.abc import Mapping
-except ImportError:
-    from collections import Mapping
-
-class DWError(RuntimeError):
-    """Interpret error number returned from dll"""
-    errors = {0: "status OK",
-              1: "error in DLL",
-              2: "cannot open d7d file",
-              3: "file already in use",
-              4: "d7d file corrupt",
-              5: "memory allocation",
-              6: "creating uncompressed file",
-              7: "extracting data",
-              8: "opening uncompressed file",
-              9: "invalid intermediate buffer level (ib_level)",
-              10: "CAN not supported",
-              }
-
-    def __init__(self, value):
-        error = self.errors.get(value, 'Unknown error {}'.format(value))
-        super(DWError, self).__init__(error)
-
-
-class DWInfo(ctypes.Structure):
-    """Structure to hold metadata about DWFile"""
+import platform
+import atexit
+import numpy as np
```

Comment:
Since this PR also contains a fair bit of refactoring, let's take the opportunity to also comply with https://peps.python.org/pep-0008/#imports, i.e. separate stdlib imports from 3rd party imports.